executor/cook/progress.py

```diff
@@ -231,41 +252,50 @@ def track_progress(progress_watcher, progress_complete_event, send_progress_upda
         progress_complete_event.set()
 
 
-def launch_progress_tracker(progress_watcher, progress_updater):
-    """Launches the threads that track progress and send progress updates to the driver.
+class ProgressTracker(object):
+    """Helper class to track progress messages from the specified location."""
 
-    Parameters
-    ----------
-    progress_watcher: ProgressWatcher
-        The progress watcher which maintains the current progress state.
-    progress_updater: ProgressUpdater
-        The progress updater which sends progress updates to the scheduler.
-        
-    Returns
-    -------
-    The progress complete threading.Event.
-    """
-    progress_complete_event = Event()
-    progress_update_thread = Thread(target=track_progress,
```

Comment:
Why isn't `track_progress` part of `ProgressTracker`?

Reply:
Will move.

Reply:
Moved as a static method inside `ProgressTracker` to keep testing simple.

Reply:
It still feels awkward to me.  We seem to be mixing OO and procedural style.

Reply:
Made `track_progress` non-static and removed the unit test. We have other tests that cover this function.